src/m365/spo/commands/site/site-recyclebinitem-clear.spec.ts

```diff
@@ -0,0 +1,192 @@
+import { Logger } from '../../../../cli/Logger';
+import Command from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./site-recyclebinitem-clear');
+
+describe(commands.SITE_RECYCLEBINITEM_CLEAR, () => {
+
+  let log: any[];
+  let logger: Logger;
+  let promptOptions: any;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.SITE_RECYCLEBINITEM_CLEAR);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the webUrl option is not a valid SharePoint site URL', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'foo', confirm: true } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the webUrl option is a valid SharePoint site URL', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com', confirm: true } }, commandInfo);
+    assert(actual);
+  });
+
+  it('prompts before removing the items from the recycle bin when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com'
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the items from the recycle bin when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'post');
```

Comment:
In this case it's better to create a stub.
Imagine if we break the command code so it doesn't abort the command, this test will timeout because the request was not stubbed.
If we write it as a stub now, we will get a clear test message saying that `postSpy.notCalled` is `false` while it should be `true`